### DISCUSSION TITLE
Deep learning feature extraction using ImageMagick?

### DISCUSSION THREAD

**Discussion by bbence84** [ImageMagick/ImageMagick#5276] [Help] [answered]
I am planning a deep learning project using photos of handwritings. I am still at the early stages, but I expect that accuracy of this neural net would improve and would be much faster if I took the larger resolution photo of the paper containing handwriting, and create a "matrix" of smaller crops containing equally sizes chunks (squares) of the original photo. Basically like cutting up a paper with scissors for equally sized pieces. Eventually, it would also be good to "normalize" e.g. the letter sizes so that each chunk would contain around the same amount of text and letter size, but that is already beyond ImageMagick I suppose.
The question is if it's possible to do a script that would take an existing image and output these small chunks, e.g. each 256x256 size max (and the "leftovers" that are on the sides are discarded). 
Thanks!

**Comment by snibgo** [comment, marked as answer]
```
magick in.png -crop 256x256 +repage out.png
```
If the width (or height) isn't a multiple of 256, you will get smaller tiles from the right (or bottom).

"+repage" removes canvas metadata, which says where each tile came from. If you want that metadata, remove "+repage".

**Reply by bbence84** [reply]
Works very good, thanks! :)
